File: Animation/app/src/main/java/com/example/animation/MainActivity.java
package com.example.animation;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.ImageView;

public class MainActivity extends AppCompatActivity {

    boolean flag = true;

    public void animate(View view){

        ImageView image1 = findViewById(R.id.imageView);
        ImageView image2 = findViewById(R.id.imageView2);

        if(flag) {
            flag = false;
            image1.animate().alpha(0).setDuration(3000);

            image2.animate().alpha(1).setDuration(3000);

        }
        else {
            flag = true;

            image1.animate().alpha(1).setDuration(3000);

            image2.animate().alpha(0).setDuration(3000);
        }
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }
}

File: DiceGame/app/src/main/java/com/example/dicegame/MainActivity.kt
package com.example.dicegame

import android.media.AudioManager
import android.media.MediaParser
import android.media.MediaPlayer
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.ImageView

class MainActivity : AppCompatActivity() {

    private fun rollDice(){

        val imageView : ImageView = findViewById(R.id.imageView)

        val diceNum = (1..6).random()

        val image = when(diceNum){

            1 -> R.drawable.dice_1
            2 -> R.drawable.dice_2
            3 -> R.drawable.dice_3
            4 -> R.drawable.dice_4
            5 -> R.drawable.dice_5
            else -> R.drawable.dice_6
        }


        imageView.animate().rotationBy(1800F).setDuration(1000)
        val mediaPlayer : MediaPlayer = MediaPlayer.create(this,R.raw.dice_sound)
        mediaPlayer.start()
        imageView.setImageResource(image)

        imageView.contentDescription = image.toString()
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val button : Button = findViewById(R.id.button)

        button.setOnClickListener {
            rollDice()
        }
    }
}